src/browser/core_state.ts

```diff
@@ -13,92 +13,110 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 /*
-	src/browser/core_state.js
-*/
+* TODO: Remove these after Dependency Injection refactor:
+* const manifestProxySettings
+* const startManifest
+* function getManifestProxySettings
+* function getStartManifest
+* function setManifestProxySettings
+* function setStartManifest
+* */
 
-// built-in modules
-const electronApp = require('electron').app;
+import * as minimist from 'minimist';
+import { app } from 'electron';
+import { ExternalApplication } from './api/external_application';
+import { PortInfo } from './port_discovery';
+import * as Shapes from '../shapes';
 
-// npm modules
-const minimist = require('minimist');
+export interface StartManifest {
+    data: Shapes.Manifest;
+    url: string;
+}
 
-// local modules
-import {
-    ExternalApplication
-} from './api/external_application';
+interface ProxySettingsArgs {
+    proxyAddress?: string;
+    proxyPort?: number;
+    type?: string;
+}
 
-// locals
-const args = electronApp.getCommandLineArguments(); // command line string ("s" for "string")
-const argv = electronApp.getCommandLineArgv(); // argument list ("v" for "vector")
-const argo = minimist(argv); // minimist-style object ("o" for "object"; hash of command line options:values; see https://github.com/substack/minimist)
+interface StrippedApplication {
```

Comment:
naming? Are these things that are found on the application, meaning is this really a subset of application?